src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/CompileFails.cs

```diff
@@ -95,6 +95,218 @@ public static IEnumerable<object[]> InvalidUnmanagedToManagedCodeSnippetsToCompi
+            {
+                ID(),
+                codeSnippets.DerivedWithStringMarshalling(utf8Marshalling, utf16Marshalling, utf8Marshalling),
+                MismatchesWithLocations(1, 2)
+            };
+            yield return new object[]
+            {
+                ID(),
+                codeSnippets.DerivedWithStringMarshalling(utf8Marshalling, utf16Marshalling, customUtf16Marshalling),
+                MismatchesWithLocations(1, 2)
+            };
+
+            // Base has no StringMarshalling and Derived does is okay
+            string source = $$"""
+                using System;
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+                namespace Test
+                {
+                    [GeneratedComInterface]
+                    [Guid("0E7204B5-4B61-4E06-B872-82BA652F2ECA")]
+                    internal partial interface INoStringMarshalling
+                    {
+                        public int GetInt();
+                        public void SetInt(int value);
+                    }
+                    [GeneratedComInterface(StringMarshalling = StringMarshalling.Utf8)]
+                    [Guid("0E7204B5-4B61-5E06-B872-82BA652F2ECA")]
+                    internal partial interface IStringMarshalling : INoStringMarshalling
+                    {
+                        public string GetString();
+                        public void SetString(string value);
+                    }
+                }
+                """;
+            yield return new object[] { ID(), source, emptyDiagnostics };
+            // Compilation can't find MarshalAs or MarshalUsing
+            //source = $$"""
+            //    using System;
+            //    using System.Runtime.InteropServices;
+            //    using System.Runtime.InteropServices.Marshalling;
+            //    namespace Test
+            //    {
+            //        [GeneratedComInterface]
+            //        [Guid("0E7204B5-4B61-4E06-B872-82BA652F2ECA")]
+            //        internal partial interface INoStringMarshalling
+            //        {
+            //            [return: MarshalUsing(typeof(Utf8StringMarshaller)]
+            //            public string GetString();
+            //            public void SetString([MarshalUsing(typeof(Utf8StringMarshaller))] string value);
+            //        }
+            //        [GeneratedComInterface(StringMarshalling = StringMarshalling.Utf16)]
+            //        [Guid("0E7204B5-4B61-5E06-B872-82BA652F2ECA")]
+            //        internal partial interface IStringMarshalling : INoStringMarshalling
+            //        {
+            //            public string GetString2();
+            //            public void SetString2(string value);
+            //        }
+            //    }
+            //    """;
+            //yield return new object[] { ID(), source, emptyDiagnostics };
```

Comment:
What's the issue with this test? Can we either fix this or remove the commented-out code?

Reply:
Ah this was an issue with copying the parameter attributes as written in the source code to the unreachable exception stub. I'll push a fix to just avoid copying the attributes.